tabrepo/benchmark/task/openml/task_utils.py

```diff
@@ -35,7 +144,10 @@ def get_ag_problem_type(task: OpenMLSupervisedTask) -> str:
     return problem_type
 
 
-def get_task_with_retry(task_id: int, max_delay_exp: int = 8) -> OpenMLSupervisedTask:
+def get_task_with_retry(task_id: int, s3_dataset_cache: str = None, max_delay_exp: int = 8) -> OpenMLSupervisedTask:
+    # Try to download from S3 first
+    download_task_from_s3(task_id, s3_dataset_cache=s3_dataset_cache)
```

Comment:
better practice to do `if s3_dataset_cache is not None:`

Otherwise someone reading the code would need to look at `download_task_from_s3` even when it isn't relevant.